File: 3팀/이가은/week19/assignment/BJ1806.java
package week19.assignment;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class BJ1806 {
    public static int min_len = 100000;
    public static int curr_len = 1;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        int s = Integer.parseInt(input[1]);

        String[] nums = br.readLine().split(" ");
        for (int i=0; i<n-1; i++){      // i = 0~n-2
            if (s <= Integer.parseInt(nums[i])) {
                min_len = 1;
                break;
            }
            for (int j=i; j<n; j++) {   // j = i~n-1
                curr_len += 1;
            }
        }
    }
}

File: 3팀/이가은/week19/assignment/BJ1806.py
import sys
sys.setrecursionlimit(10001)

n, s = map(int, input().split())
nums = list(map(int, input().split()))
min_len = 100000

for i in range(n-1): # i = 0~n-2
    length = 1
    sum = nums[i]
    if sum >= s:
        min_len = 1
        break

    for j in range(i, n, 1):    # j = i~n-1
        sum += nums[j]
        length += 1
        if sum >= s:
            min_len = min(length, min_len)
            break

if min_len == 10000:
    print(0)
else:
    print(min_len)

